ambari-server/src/main/java/org/apache/ambari/server/agent/stomp/dto/MetadataCluster.java

```diff
@@ -67,44 +82,58 @@ public static MetadataCluster emptyMetadataCluster() {
     return clusterLevelParams;
   }
 
-  public void setClusterLevelParams(SortedMap<String, String> clusterLevelParams) {
-    this.clusterLevelParams = clusterLevelParams;
-  }
-
   public SortedMap<String, SortedMap<String, String>> getAgentConfigs() {
     return agentConfigs;
   }
 
-  public void setAgentConfigs(SortedMap<String, SortedMap<String, String>> agentConfigs) {
-    this.agentConfigs = agentConfigs;
+  public boolean isFullServiceLevelMetadata() {
+    return fullServiceLevelMetadata;
   }
 
-  public boolean updateServiceLevelParams(SortedMap<String, MetadataServiceInfo> update) {
-    boolean changed = false;
-    for (String key : update.keySet()) {
-      if (!serviceLevelParams.containsKey(key) || !serviceLevelParams.get(key).equals(update.get(key))) {
-        changed = true;
-        break;
+  public boolean updateServiceLevelParams(SortedMap<String, MetadataServiceInfo> update, boolean fullMetadataInUpdatedMap) {
+    if (update != null) {
+      if (this.serviceLevelParams == null) {
+        this.serviceLevelParams = new TreeMap<>();
```

Comment:
I think these map creations should also be guarded by `lock`.

Reply:
You are right (this was a last minute movement at certainly not thought over); it's fixed now.